production/helm/loki/templates/statefulset.yaml

```diff
@@ -83,11 +81,25 @@ spec:
         - name: config
           secret:
             secretName: {{ template "loki.fullname" . }}
+  {{- if not .Values.persistence.enabled }}
         - name: storage
-      {{- if .Values.persistence.enabled }}
-          persistentVolumeClaim:
-            claimName: {{ .Values.persistence.existingClaim | default (include "loki.fullname" .) }}
-      {{- else }}
           emptyDir: {}
-      {{- end }}
+  {{- else if .Values.persistence.existingClaim }}
+        - name: storage
+          persistentVolumeClaim:
+            claimName: {{ .Values.persistence.existingClaim }}
```

Comment:
does this work with more than one replica - so they all use the same existingClaim?

Reply:
Yes, if that persistent volume bound to the PVC supports `ReadWriteMany` access mode like NFS. It us up to the user to provision the PV and PVC.

Reply:
Yes, but I wonder if the application (loki) does support that ...

Reply:
The chunk filenames are uuid's, so writes in the same dir do not collide. At least haven't seen any issues yet with 5k lines per sec in my setup.

Reply:
ok